jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -195,6 +201,45 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    @with_spinner()
+    def cone_aperture(self):
```

Comment:
how does this work within `collapse_to_spectrum`?  The end product should be the 1d spectrum itself, so I don't think we need to worry about visualizing the mask aside from the cone representation in #2664... or does that need to be modified to differentiate between the different methods -  exact, subpixel, center - being proposed here?

Reply:
I think this is also a question for @bmorris3 . I was able to get fractional pixel support for a cone aperture going through the cube. Neither Jira tickets associated with this effort explain what the next step is, so do you have a suggestion on how to move forward here? Should the cone aperture be used to create a 1d spectrum in `collapse_to_spectrum`? Should it update the marks in #2664 ? Some direction here would be very appreciated!

Reply:
Currently `collapse_to_spectrum` converts the cube data entry (and optionally crop it to data within _N_ subsets) to a spectrum. I think the best organization going forward is to add a condition [in this block](https://github.com/spacetelescope/jdaviz/blob/d61227615101f129f89d1359d61a9428ede61b4f/jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py#L130-L139) to return a conically-weighted spectral cube for the collapsing operations.

Reply:
@kecnry – I don't think we should change the marks according to the method. I do think that we should put a [link to the relevant photutils docs](https://photutils.readthedocs.io/en/stable/aperture.html#aperture-masks) in the UI to define those method names.

Reply:
Am I only using `aperture.selected` to create the conically-weighted spectral cube or do I include the background subset as well?

Reply:
this is coming before background subtraction, so no need to implement for the background itself, yet, but try to keep that in mind (that we'll likely need to re-use the same code for the background as well).